cypress/e2e/models/migration/dynamic-report/insights/insights.ts

```diff
@@ -0,0 +1,66 @@
+import {
+    clickByText,
+    getUniqueElementsFromSecondArray,
+    validateAnyNumberPresence,
+    validateTextPresence,
+} from "../../../../../utils/utils";
+import { DynamicReports } from "../../../../tests/migration/dynamic-report/dynamic-report";
+import { issueFilter, trTag } from "../../../../types/constants";
+import { AppInsight } from "../../../../types/types";
+import { liTag, span } from "../../../../views/common.view";
+import { insightColumns, singleApplicationColumns } from "../../../../views/issue.view";
+
+export class Insights extends DynamicReports {
+    static urlSuffix = "/insights";
+    static menuName = "Insights";
+
+    public static applyAndValidateFilter(
+        filterType: issueFilter,
+        filterValues: string[],
+        insightsExpected: AppInsight[],
+        insightsNotExpected?: AppInsight[],
+        isSingle = false
+    ) {
+        filterValues.forEach((value) => {
+            Insights.applyFilter(filterType, value, isSingle);
+        });
+        insightsExpected.forEach((insight) => {
+            Insights.validateFilter(insight, isSingle);
+        });
+
+        if (insightsNotExpected.length > 0) {
```

Comment:
insightsNotExpected is optional. If the caller does not pass it, this line will throw error.

You can do : 

`if (insightsNotExpected && insightsNotExpected.length > 0) {`